src/please.go

```diff
@@ -864,6 +867,61 @@ var buildFunctions = map[string]func() int{
 	},
 }
 
+// Check if tool is given as label or path and then run
+func CheckIsLabelOrPathAndRun(config *core.Configuration, _tool tool.Tool, args []string) {
+//	log.Warning("Sam: Hi I'm in CheckIsLabelOrPathAndRun()")
+	tools := tool.MatchingTools(config, string(_tool))
+//	log.Warning("Sam: tools=%s", tools)
+	target := fs.ExpandHomePath(tools[tool.AllToolNames(config, string(_tool))[0]])
+	label := core.ParseBuildLabels([]string{target})
+//	log.Warning("Sam: target=%s", target)
+	if !core.LooksLikeABuildLabel(target) {
+//		log.Warning("Sam: Code path 1")
+		if !filepath.IsAbs(target) {
+			t, err := core.LookBuildPath(target, config)
+			if err != nil {
+				log.Fatalf("%s", err)
+			}
+			target = t
+		}
+		// Hopefully we have an absolute path now, so let's run it.
+		if success, state := runBuild(label, true, false, false); success {
```

Comment:
We don't need to run a build if the tool isn't a build label. We can just do `tool.Run(config, "build_langserver", opts.Tool.Args.Args.AsStrings())`. `tool.Run()` can assume the tool is either an abs path to a binary, or it's a file on the path. IF we move the above logic out into `too.Run()`, that should be fine.